runtime/src/sdk.rs

```diff
@@ -113,6 +113,20 @@ pub unsafe extern "C" fn extism_plugin_destroy(plugin: PluginIndex) {
     if p.is_some() {
         *p = None
     }
+
+    // Attempt to cleanup if there are `None`s at the end of the vec
+    for (i, x) in plugins.iter().rev().enumerate() {
+        let i = plugins.len() - i;
+
+        // Stop at first `Some`
+        if x.is_some() {
+            if i < plugins.len() {
+                plugins.truncate(i)
+            }
+
+            break;
+        }
+    }
 }
```

Comment:
Nice! Before we merge this, I may add some tests. Now that there is some destructive behavior I think it would be good to have a few checks around what is destroyed/retained and verify that there are no side effects.